src/cube_blog.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+
+import BlogPost from './components/BlogPost';
+import Paginate from './components/Paginate';
+
+class CubeBlog extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+        posts: this.props.posts,
+        pages: this.props.pages,
+        canEdit: this.props.canEdit
+      };
+  }
+
+  componentDidMount() {
+    updateBlog();
+  }
+
+  select(nav) {
+    this.setState({ });
+  }
+
+  render() {
```

Comment:
destructuring props would help here `const {pages, posts, canEdit} = this.props` means you can change these variables from `this.props.VAR` to `VAR` in the rest of the file